### PR TITLE
chore: add angular standalone components blog post

### PR DESCRIPTION
**Pull request by ankypant**
### What is this article about?
R: Angular standalone components

### Checklist:

- [x] I am added to the list of authors correctly.
- [x] All images used in the post is in the correct folder (`assets/images/post/`).
- [x] My article has been reviewed internally before opening this Pull Request.
- [x] I have not modified someone else's previously published articles.
- [x] I double checked my changes locally at http://127.0.0.1:4000/ after running `bundle install && bundle exec jekyll serve`.
- [x] My changes generate no new errors

### CODE REVIEW

File: assets/images/post/angular-standalone-components.png
Comment (ankypant) [inline assets/images/post/angular-standalone-components.png:1]: We might have to replace the banner, I got this from the Angular web-site

Comment (arthurpalves) [reply assets/images/post/angular-standalone-components.png:1]: Looking into it

**Review by santam85 — APPROVED** [review]
Nice one @ankypant !
Consider addressing some of the warnings from the Vale action, but ok as it is!

File: _posts/2023-06-26-angular-standalone-components.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+- Ankit Pant
+header:
+  teaser: /assets/images/post/angular-standalone-components.png
+  teaser_alt: Angular standalone components
+category: Frontend
+---
+
+![](/assets/images/post/angular-standalone-components.png)
+
+# Introduction
```
Comment (tcorral) [inline _posts/2023-06-26-angular-standalone-components.md]: I miss saying on which Angular version the standalone component was introduced and what are the benefits of using standalone components.

File: _posts/2023-06-26-angular-standalone-components.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+# Migrate a component to a Standalone
+
+You can convert a component in one of the following ways:
+
+* Using the [schematics](https://angular.io/guide/standalone-migration#migrate-an-existing-angular-project-to-standalone "https://angular.io/guide/standalone-migration#migrate-an-existing-angular-project-to-standalone")
+
+* Manually
+
+## Migrate to Standalone via schematics
```
Comment (tcorral) [inline _posts/2023-06-26-angular-standalone-components.md]: I don't know where should I execute those commands? In the root of the project or in the folder of the module I want to migrate? As a suggestion, I think we should recommend having the code covered with unit tests and e2e tests before the migration, if it's possible.

File: _posts/2023-06-26-angular-standalone-components.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+description="Specs for Standalone component"
+%}
+
+_**NOTE:**_ After the migration the specs might be broken and might need manual fixes
+
+# Improvements on performance using standalone feature
+
+Here are a few ways in which Angular standalone components can contribute to performance improvements:
+
+* **Reduced bundle size:** You can reduce the overall bundle size of your app by using Angular's standalone feature and eliminating NgModules and the boilerplate code that comes with it.
```
Comment (tcorral) [inline _posts/2023-06-26-angular-standalone-components.md]: Is the "Reduced bundle size" correct? I understood from Kirill that is not always true. but I liked you added the comparison.

File: _posts/2023-06-26-angular-standalone-components.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+* **Isolation and lazy loading:** You can enhance the performance by encapsulating functionality within standalone components which helps to load all the resource only when it is lazy loaded.
+
+* **Reusability:** Angular standalone components are self-contained, independent, and can be re-utilized across the application.
```
Comment (tcorral) [inline _posts/2023-06-26-angular-standalone-components.md]: Is there any change in how to package and distribute a standalone component compared to the NgModule ones?

File: _posts/2023-06-26-angular-standalone-components.md (hunk 1 above)
Comment (ankypant) [reply _posts/2023-06-26-angular-standalone-components.md]: @tcorral I can add this to the introduction but the benefits are described in the performance section.

File: _posts/2023-06-26-angular-standalone-components.md (hunk 2 above)
Comment (ankypant) [reply _posts/2023-06-26-angular-standalone-components.md]: Yes, this should be executed in the root of an application and of course, the app developers are responsible for making sure that the application is already tested and works before the migration.
@tcorral Do you want me to add this information as well?

File: _posts/2023-06-26-angular-standalone-components.md (hunk 3 above)
Comment (ankypant) [reply _posts/2023-06-26-angular-standalone-components.md]: Yeah, it depends from application to application but according to me, we are reducing the boilerplate from the code base by getting rid of NgModules.

File: _posts/2023-06-26-angular-standalone-components.md (hunk 4 above)
Comment (ankypant) [reply _posts/2023-06-26-angular-standalone-components.md]: @tcorral There is no change in packaging and distribution with standalone components.
This point highlights that standalone components are self-contained and could be imported directly into other components without having to import the dependencies again and again like we use to do with NgModules

File: _posts/2023-06-26-angular-standalone-components.md (hunk 1 above)
Comment (ankypant) [reply _posts/2023-06-26-angular-standalone-components.md]: @tcorral I have updated the documentation for the required information starting from line 18.

File: _posts/2023-06-26-angular-standalone-components.md (hunk 2 above)
Comment (ankypant) [reply _posts/2023-06-26-angular-standalone-components.md]: Updated the document with your suggestion with the Prerequisites and a note for running the schematics command at the root of the project.

File: assets/images/post/angular-standalone-components.png
Comment (ankypant) [reply assets/images/post/angular-standalone-components.png:1]: Hello @arthurpalves can you please help me with the banner :)

**Review by tcorral — APPROVED** [review]

**Review by robinsonzimmermann — APPROVED** [review]

**Pull request merged**